src/main/java/org/testng/reporters/util/StackTraceTools.java

```diff
@@ -50,4 +50,15 @@ public static int getTestRoot(StackTraceElement[] stack, ITestNGMethod method) {
       return new StackTraceElement[0];
     }
   }
+  
+  /***
+   * Given a trace element, format it's line number + fully qualified
+   * class name, like you would see in a stack trace.
+   */
+  public static String formatByStackTrace(StackTraceElement trace) {
+    String[] splits = trace.getClassName().split("\\.");
+    String className = splits[splits.length-1];
+    String fullClassName = trace.getClassName();
+    return " at " + fullClassName + "." + trace.getMethodName() + "(" + className + ".java:" + trace.getLineNumber() + ")";
```

Comment:
This should end with a newline `\n` (or) at-least a space, so that if there are multiple soft assertion failures the lines will appear properly. The current implementation just appends everything together.